identity-service/src/routes/trackListens.js

```diff
@@ -315,19 +313,21 @@ module.exports = function (app) {
     const isSolanaListenEnabled = getFeatureFlag(optimizelyClient, FEATURE_FLAGS.SOLANA_LISTEN_ENABLED_SERVER)
     const solanaListen = req.body.solanaListen || isSolanaListenEnabled || false
 
-    let currentHour = await getListenHour()
+    const currentHour = trimToHour(new Date())
     // Dedicated listen flow
     if (solanaListen) {
-      let suffix = currentHour.toISOString()
+      const suffix = currentHour.toISOString()
+      const entropy = uuidv4()
 
       // Example key format = listens-tx-success::2022-01-25T21:00:00.000Z
-      let trackingRedisKeys = getTrackingListenKeys(suffix)
+      const trackingRedisKeys = getTrackingListenKeys(suffix)
       await initializeExpiringRedisKey(redis, trackingRedisKeys.submission, redisTxTrackingExpirySeconds)
       await initializeExpiringRedisKey(redis, trackingRedisKeys.success, redisTxTrackingExpirySeconds)
 
       req.logger.info(`TrackListen tx submission, trackId=${trackId} userId=${userId}, ${JSON.stringify(trackingRedisKeys)}`)
 
       await redis.incr(trackingRedisKeys.submission)
+      await redis.zadd(TRACKING_LISTEN_SUBMISSION_KEY, Date.now(), Date.now() + entropy)
```

Comment:
is this entropy just to guarantee no collisions?

Reply:
Yes exactly. It's a recommended best practice -- https://redis.com/redis-best-practices/time-series/sorted-set-time-series/